adminapi/dataset/__init__.py

```diff
@@ -269,7 +276,7 @@ def __init__(self, attributes=[]):
             if isinstance(value, (tuple, list, set, frozenset)):
                 attributes[attribute_id] = MultiAttr(value, self, attribute_id)
         super(DatasetObject, self).__init__(attributes)
-        self.object_id = self.get('object_id')
+        self.object_id = object_id
```

Comment:
Shouldn't this be `_object_id`?  The external code should access `obj['object_id']`.

Reply:
obj['object_id'] is only there if 'object_id' was in restrict, while this object_id will always be there if the object is not new or deleted. We access that attribute in the graphite cache as well as in the servershell which makes it not really internal.